### DISCUSSION TITLE
How the QOS values transferring from ROS to DDS layer

### DISCUSSION THREAD

**Discussion by aqnaazeez** [eProsima/Fast-DDS#4909] [Q&A]
I am analysing how the DDS(eprosima FastDDS) Qos values change with respect to the ROS2 humble QOS settings. While changing the QOS settings at ROS2 publisher application the QoS values in the DDS layer (by added print statements to the DataWriterQos and DataReaderQos of FastDDS) always printing default QoS values. 
1. Where should I expect these changes to be reflected in the DDS layer? 
2. Which ros2 rmw_fastrtps function is responsible for transferring this QOS settings to DDS. Similarly which DDS function is receiving the Qos settings from ROS2?
3. Currently checking with the QOS policies Reliability, Durability, History and liveliness. In which all DDS layers these policies will be effected?

**Comment by JesusPoderoso** [comment]
Hi @aqnaazeez, thanks for using Fast DDS.
There is an [specific section](https://github.com/ros2/rmw_fastrtps?tab=readme-ov-file#full-qos-configuration) in the [rmw_fastrtps repository README](https://github.com/ros2/rmw_fastrtps) with some guidelines regarding those QoS configurations.
